### PR TITLE
[tests-only][full-ci]Assign role using graph api

### PR DESCRIPTION
**Pull request by amrita-shrestha**
## Description
This PR adds a method to assign roles to user using graph api 

## Related Issue

- Fixes <https://github.com/owncloud/web/issues/8325>

## How Has This Been Tested?

- Locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

**Review by SagarGi — COMMENTED** [review]
LGTM :+1: if CI is green. I think you can also clean up the settings if its not used by other steps.

File: tests/e2e/support/api/graph/userManagement.ts [hunk 1]
```diff
@@ -129,3 +137,23 @@ export const addUserToGroup = async ({
...
+
+  const applicationEntities = await getAppEntities(admin)
+  const response = await request({
+    method: 'POST',
+    path: join('graph', 'v1.0', 'users', id, 'appRoleAssignments'),
+    user: admin,
+    body: JSON.stringify({
+      principalId: id,
+      appRoleId: roleId,
+      resourceId: applicationEntities[0].id
```
Comment (saw-jan) [inline tests/e2e/support/api/graph/userManagement.ts:155]: use of index can be problematic later
```suggestion
      resourceId: applicationEntities[0].id
```

**Review by saw-jan — COMMENTED** [review]
looks good

File: tests/e2e/support/api/graph/userManagement.ts [hunk 2]
```diff
@@ -2,6 +2,14 @@ import { checkResponseStatus, request } from '../http'
...
 import join from 'join-path'
 import { config } from '../../../config'
+import { getApplicationEntity } from './utils'
+
+enum Roles {
+  SpaceAdmin = '2aadd357-682c-406b-8874-293091995fdd',
+  Admin = '71881883-1768-46bd-a24d-a356a2afdf7f',
+  Guest = '38071a68-456a-4553-846a-fa67bf5596cc',
+  User = 'd7beeea8-8ff4-406b-8fb6-ab2dd81e6b11'
+}
```
Comment (SwikritiT) [inline tests/e2e/support/api/graph/userManagement.ts:12]: Do this change or are static for everyone?

Comment (saw-jan) [inline tests/e2e/support/api/graph/userManagement.ts:12]: getApplicationEntity returns this information, maybe you could get role-ids from there instead of hardcoded.

Comment (saw-jan) [reply tests/e2e/support/api/graph/userManagement.ts:12]: No idea whether these values will change or not
 
CC @ScharfViktor

Comment (ScharfViktor) [reply tests/e2e/support/api/graph/userManagement.ts:12]: I think the IDs can be changed in the future. The best solution seems to me to get the roles from  GET `/graph/v1.0/applications`
Screenshot 2023-02-08 at 09 25 06

Comment (saw-jan) [reply tests/e2e/support/api/graph/userManagement.ts:12]: ^ @amrita-shrestha

File: tests/e2e/support/api/graph/userManagement.ts [hunk 3]
```diff
@@ -129,3 +131,27 @@ export const addUserToGroup = async ({
...
   checkResponseStatus(response, 'Failed while adding an user to the group')
 }
+
+export const assignRole = async (admin: User, id: string, role: string): Promise<void> => {
+  const applicationEntity = await getApplicationEntity(admin)
+
+  if (!roleStore.has(role)) {
+    applicationEntity.appRoles.map((key, value) => {
+      return roleStore.set(key.displayName, key.id)
+    })
```
Comment (saw-jan) [inline tests/e2e/support/api/graph/userManagement.ts:141]: if you are not using the array returned by map
```suggestion
    applicationEntity.appRoles.forEach((role) => {
      roleStore.set(rome.displayName, role.id)
    })
```

File: tests/e2e/support/store/role.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+export const roleStore = new Map<string, string>()
```
Comment (saw-jan) [inline tests/e2e/support/store/role.ts:1]:
```suggestion
export const userRoleStore = new Map<string, string>()
```

**Review by saw-jan — COMMENTED** [review]
Others looks good

File: tests/e2e/cucumber/features/smoke/admin-settings/spaces.ocis.feature [hunk 5]
```diff
@@ -6,9 +6,9 @@ Feature: spaces management
       | Alice |
       | Brian |
     And "Admin" assigns following roles to the users
```
Comment (ScharfViktor) [inline tests/e2e/cucumber/features/smoke/admin-settings/spaces.ocis.feature:8]: only comment, we can do it in separate PR.
maybe change it as: `And "Admin" assigns following roles to the users using API`?
and change another steps from https://github.com/owncloud/web/blob/master/tests/e2e/cucumber/steps/api.ts

File: tests/e2e/support/api/graph/utils.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+interface ApplicationEntity {
+  appRoles: AppRole[]
+  displayName: string
+  id: string
+}
+
+interface AppRole {
+  displayName: string
+  id: string
+}
```
Comment (ScharfViktor) [inline tests/e2e/support/api/graph/utils.ts:14]: I didn't find where we use this interface

**Comment by ScharfViktor** [comment]
looks cool. thanks for redoing it to use the new graph methods :heart:

File: tests/e2e/support/api/graph/utils.ts (hunk 6 above)
Comment (amrita-shrestha) [reply tests/e2e/support/api/graph/utils.ts:14]: just above this interface line 6
```
interface ApplicationEntity {
  appRoles: AppRole[]
  displayName: string
  id: string
}
```

Comment (saw-jan) [reply tests/e2e/support/api/graph/utils.ts:14]: used in the above interface `L:6`
```ts
interface ApplicationEntity {
  appRoles: AppRole[]
  displayName: string
  id: string
}
```

**Review by ScharfViktor — APPROVED** [review]
LGTM
